src/gsy_e/models/area/__init__.py

```diff
@@ -240,17 +242,88 @@ def update_descendants_strategy_prices(self):
             log.exception("area.update_descendants_strategy_prices failed.")
             return
 
-    def cycle_coefficients_trading(self, current_time_slot: DateTime) -> None:
-        """Perform operations that should be executed on coefficients trading cycle."""
-        self._current_time_slot = current_time_slot
-
     def get_results_dict(self):
         """Calculate the results dict for the coefficients trading."""
         if self.strategy is not None:
-            return self.strategy.state.get_results_dict(self._current_time_slot)
+            return self.strategy.state.get_results_dict(self._current_market_time_slot)
         return {}
 
 
+class CoefficientArea(AreaBase):
+    """Area class for the coefficient matching mechanism."""
+    def __init__(self, name: str = None, children: List["CoefficientArea"] = None,
+                 uuid: str = None,
+                 strategy: SCMStrategy = None,
+                 config: SimulationConfig = None,
+                 grid_fee_percentage: float = None,
+                 grid_fee_constant: float = None,
+                 coefficient_percent: float = 0.0,
+                 market_maker_rate_eur: float = (
+                         ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE / 100.),
+                 feed_in_tariff_eur: float = GlobalConfig.FEED_IN_TARIFF / 100.,
```

Comment:
Since this is on the datasheet then I will change it, however to my knowledge this regulation is only for European countries and will not apply to any of the other currencies (might be for Switzerland but I am not certain even for this). Nevertheless, does not hurt to remove euro from the name. I will also add comments to indicate that these are in the currency of each respective country that is used.